lambdas/ec2/reaper.py

```diff
@@ -231,25 +267,30 @@ def terminate_expired_instances(event, context):
         ec2_termination_date = get_tag(instance, 'termination_date')
         if ec2_termination_date is None:
             print("No termination date found for {0}".format(instance.id))
+            stop_instance(instance, "EC2 instance has no termination_date")
             improperly_tagged.append(instance)
             continue
-        try:
-            dateutil.parser.parse(ec2_termination_date) - timenow_with_utc()
-        except Exception as e:
-            print("Unable to parse the termination_date for {0}".format(instance.id))
-            improperly_tagged.append(instance)
+        if ec2_termination_date != INDEFINITE:
```

Comment:
I think that this conditional should be combined with the exact same conditional below; it saves a redundant operation, and will be a bit cleaner in determining the logic:

```python
        if ec2_termination_date != INDEFINITE:
                try:            
                  if dateutil.parser.parse(ec2_termination_date) > timenow_with_utc():
                    ttl = dateutil.parser.parse(ec2_termination_date) - timenow_with_utc()
                     print("EC2 instance will be terminated {0} seconds from now, roughly".format(ttl.seconds))
                  else:
                    terminate_instance(instance, "EC2 instance is expired")
                    deleted_instances.append(instance)
                except Exception as e:
                  print("Unable to parse the termination_date for {0}".format(instance.id))
                  stop_instance(instance, "EC2 instance has invalid termination_date")
                  improperly_tagged.append(instance)
                  continue

Reply:
Done!